engine/api/objectstore/fs.go

```diff
@@ -12,20 +12,31 @@ import (
 
 // FilesystemStore implements ObjectStore interface with filesystem driver
 type FilesystemStore struct {
-	basedir string
+	projectIntegration sdk.ProjectIntegration
+	basedir            string
 }
 
-// NewFilesystemStore creates a new ObjectStore with filesystem driver
-func NewFilesystemStore(basedir string) (*FilesystemStore, error) {
+// newFilesystemStore creates a new ObjectStore with filesystem driver
+func newFilesystemStore(projectIntegration sdk.ProjectIntegration, basedir string) (*FilesystemStore, error) {
 	log.Info("Objectstore> Initialize Filesystem driver on directory: %s", basedir)
 	if basedir == "" {
 		return nil, fmt.Errorf("artifact storage is filesystem, but --artifact-basedir is not provided")
 	}
 
-	fss := &FilesystemStore{basedir: basedir}
+	fss := &FilesystemStore{projectIntegration: projectIntegration, basedir: basedir}
 	return fss, nil
 }
 
+// TemporaryURLSupported returns true is temporary URL are supported
+func (fss *FilesystemStore) TemporaryURLSupported() bool {
+	return false
+}
+
+// GetProjectIntegration returns current projet Integration, nil otherwise
+func (fss *FilesystemStore) GetProjectIntegration() sdk.ProjectIntegration {
```

Comment:
sdk.ProjectIntegration can be nil ?

Reply:
If a FilesystemStore exists, GetProjectIntegration() can't be nil